### PR TITLE
obb: fix compatibility with Eigen 3.0.5

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

File: src/BV/OBB.cpp [hunk 1]
```diff
@@ -317,7 +317,11 @@ bool obbDisjointAndLowerBoundDistance (const Matrix3f& B, const Vec3f& T,
   // Corner of b axis aligned bounding box the closest to the origin
   Vec3f AABB_corner (T.cwiseAbs () - Bf * b);
   Vec3f diff3 (AABB_corner - a);
+#if EIGEN_VERSION_AT_LEAST(3,1,0)
   diff3 = diff3.cwiseMax (0);
```
Comment (jmirabel) [inline src/BV/OBB.cpp]: Did you check `diff3.array().max(0)` for Eigen < 3.1.0 ? If it works, I would rather avoid the preprocessor if/else.

Comment (jcarpent) [reply src/BV/OBB.cpp]: There is not Scalar overload of `cwiseMax` or `max` with Eigen lower than 3.1.0

Comment (jmirabel) [reply src/BV/OBB.cpp]: The following code compiles fine with Eigen 3.1.0:
```c++
#include <Eigen/Dense>
#include <iostream>

int main()
{
  Eigen::Array3d A(0,1,2);
  std::cout << A.max (1) << '\n';
}
```
so I guess there is an overload in Array class.

Comment (jcarpent) [reply src/BV/OBB.cpp]: But not with 3.0.x

Comment (jcarpent) [reply src/BV/OBB.cpp]: Yes, they have added the overload in 3.1.0

Comment (jmirabel) [reply src/BV/OBB.cpp]: You are right. It does not compile with Eigen 3.0.0.

File: src/BV/OBB.cpp [hunk 2]
```diff
@@ -317,7 +317,8 @@ bool obbDisjointAndLowerBoundDistance (const Matrix3f& B, const Vec3f& T,
   // Corner of b axis aligned bounding box the closest to the origin
   Vec3f AABB_corner (T.cwiseAbs () - Bf * b);
   Vec3f diff3 (AABB_corner - a);
-  diff3 = diff3.cwiseMax (0);
+  diff3 = diff3.cwiseMax (Vec3f::Zero());
```
Comment (jmirabel) [inline src/BV/OBB.cpp]: @jcarpent  Does this suit you ?

I prefer to avoid preprocessor if/else when possible.

Comment (jcarpent) [reply src/BV/OBB.cpp]: Yes, I would have done it at first glance as I was thinking you wanted to keep the initial version.
It is perfect for me now.

Comment (jmirabel) [reply src/BV/OBB.cpp]: Both version boils down to the same assembly code so I give the priority to the most readable solution.

**Pull request merged**